pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/inv/text/LuceneTextIndexCreator.java

```diff
@@ -115,14 +115,31 @@ public IndexWriter getIndexWriter() {
 
   @Override
   public void addDoc(Object document, int docIdCounter) {
-    Document docToIndex = new Document();
-    docToIndex.add(new TextField(_textColumn, document.toString(), Field.Store.NO));
-    docToIndex.add(new StoredField(LUCENE_INDEX_DOC_ID_COLUMN_NAME, docIdCounter));
-    try {
-      _indexWriter.addDocument(docToIndex);
-    } catch (Exception e) {
-      LOGGER.error("Failure while adding a new document to index for column {}, exception {}", _textColumn, e.getMessage());
-      throw new RuntimeException(e);
+    if (!(document instanceof Object[])) {
+      // text index on SV column
+      Document docToIndex = new Document();
+      docToIndex.add(new TextField(_textColumn, document.toString(), Field.Store.NO));
+      docToIndex.add(new StoredField(LUCENE_INDEX_DOC_ID_COLUMN_NAME, docIdCounter));
+      try {
+        _indexWriter.addDocument(docToIndex);
+      } catch (Exception e) {
+        LOGGER.error("Failure while adding a new document to index for column {}, exception {}", _textColumn, e.getMessage());
+        throw new RuntimeException(e);
+      }
+    } else {
+      // text index on MV column
+      Object[] values = (Object[])document;
+      for (Object value : values) {
+        Document docToIndex = new Document();
```

Comment:
Will this cause multiple text document map to the same pinot document? How do you maintain the map? We need to figure out the semantic of text index on MV columns

Reply:
In order to fix this issue https://github.com/apache/incubator-pinot/issues/5666 sooner, I am only relaxing the constraint of text index on raw columns in this PR. The constraint of text index on SV columns is still there and will address in a follow-up. Need to do some digging into Lucene as well to see how arrays can be stored in a single document etc.